### PR TITLE
chore(broker-core): migrates TimerCatchEventTest to use static broker

### PR DESCRIPTION
**Pull request by npepinpe**

### CODE REVIEW

**Comment by npepinpe** [comment]
Extracted from ongoing issue as I used this class initially as to test changes, and migrated it to using a static embedded broker to speed it up.

File: broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java [hunk 1]
```diff
@@ -84,35 +97,36 @@
...
 
     testClient.deploy(workflow);
-    testClient.createWorkflowInstance(r -> r.setBpmnProcessId(PROCESS_ID));
+    testClient.createWorkflowInstance(r -> r.setBpmnProcessId(processId));
 
     // then
     assertThat(
             RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_COMPLETED)
-                .withElementId(PROCESS_ID)
+                .withElementId(processId)
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java]: Filter by workflow instance key.

File: broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java [hunk 2]
```diff
@@ -84,35 +97,36 @@
...
+    brokerRule.getClock().reset();
+  }
+
   @Test
   public void testLifeCycle() {
     // given
+    final String processId = Strings.newRandomValidBpmnId();
     final BpmnModelInstance workflow =
-        Bpmn.createExecutableProcess(PROCESS_ID)
+        Bpmn.createExecutableProcess(processId)
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java]: I think it is not necessary to generate a process id. To be safe, we could use the workflow key.

File: broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java [hunk 3]
```diff
@@ -211,74 +245,102 @@ public void shouldCompleteTimerEvent() {
   @Test
   public void shouldTriggerTimerWithZeroDuration() {
     // given
+    final String processId = Strings.newRandomValidBpmnId();
+    final String timerId = Strings.newRandomValidBpmnId();
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java]: No need to generate the element id if you filter the records by workflow instance key. Random id may make it harder to understand the records if the test fails.

**Review by saig0 — CHANGES_REQUESTED** [review]

**Comment by npepinpe** [comment]
Updated after review; I prefer deploying with workflow key than process. Seems redundant for those with static names, but I think its a better habit to create an instance based on the exact process you just deployed, and makes it easier in case we ever change the process for a given test.

**Comment by npepinpe** [comment]
Hi, sorry, apparently I forgot how to git and borked the code while rebasing, it should be fixed now with your review hints applied.

**Review by saig0 — APPROVED** [review]

**Pull request merged**